lib/src/core/services/networking/api_endpoint.dart

```diff
@@ -0,0 +1,123 @@
+import 'package:empriusapp/src/core/config/config.dart';
+import 'package:flutter/material.dart';
+
+/// A utility class for getting paths for API endpoints
+/// without involving strings.
+/// This class has no constructor and all methods are `static`.
+@immutable
+class ApiEndpoint {
+  const ApiEndpoint._();
+
+  /// The base url of our REST API, to which all the requests will be sent.
+  /// It is supplied at the time of building the apk or running the app:
+  /// ```
+  /// flutter build apk --debug --dart-define=BASE_URL=www.some_url.com
+  /// ```
+  /// OR
+  /// ```
+  /// flutter run --dart-define=BASE_URL=www.some_url.com
+  /// ```
+  static const baseUrl = Config.baseUrl;
+
+  /// Returns the path for an authentication [endpoint].
+  static String auth(AuthEndpoint endpoint) {
+    const path = '/auth';
+    switch (endpoint) {
+      case AuthEndpoint.REGISTER: return '$path/register';
+      case AuthEndpoint.LOGIN: return '$path/login';
+      case AuthEndpoint.REFRESH_TOKEN: return '$path/refresh-token';
+      case AuthEndpoint.CHANGE_PASSWORD: return '$path/change-password';
+      case AuthEndpoint.FORGOT_PW_SEND_OTP: return '$path/forgot/send-otp';
+      case AuthEndpoint.FORGOT_PW_VERIFY_OTP: return '$path/forgot/verify-otp';
+      case AuthEndpoint.FORGOT_PW_RESET_PASSWORD: return '$path/forgot/reset-password';
+    }
+  }
+
+  /// Enums for different endpoints collections:
+  ///
+  /// Returns the path for a user bookings [endpoint].
+  ///
+  /// Specify student connection [id] to get the path for a specific student connection.
+  static String userConnections(StudentConnectionEndpoint endpoint, {int? id}) {
```

Comment:
This is not own

Reply:
I copied the example as it looks neat to reply in our endpoints when they are defined

Reply:
Same as above, if is not used, shouldn't be commited